pkg/apis/servicecatalog/validation/instance.go

```diff
@@ -254,13 +256,31 @@ func validateServiceInstanceUpdate(instance *sc.ServiceInstance) field.ErrorList
 }
 
 // internalValidateServiceInstanceUpdateAllowed ensures there is not a
-// pending update on-going with the spec of the instance before allowing an update
-// to the spec to go through.
+// pending update on-going with the spec of the instance before allowing an
+// update to the spec to go through unless its the same user who made the
+// original update.
 func internalValidateServiceInstanceUpdateAllowed(new *sc.ServiceInstance, old *sc.ServiceInstance) field.ErrorList {
 	errors := field.ErrorList{}
-	if old.Generation != new.Generation && old.Status.CurrentOperation != "" {
-		errors = append(errors, field.Forbidden(field.NewPath("spec"), "Another update for this service instance is in progress"))
+
+	// If the OriginatingIdentityLocking feature is not set then only allow the
+	// same user to edit the Instance.
+	if utilfeature.DefaultFeatureGate.Enabled(scfeatures.OriginatingIdentityLocking) {
+		oldUID, newUID := "", ""
+
+		if old.Spec.UserInfo != nil {
+			oldUID = old.Spec.UserInfo.UID
+		}
+
+		if new.Spec.UserInfo != nil {
+			newUID = new.Spec.UserInfo.UID
+		}
+
+		if old.Generation != new.Generation && old.Status.CurrentOperation != "" && oldUID != newUID {
```

Comment:
@duglin I'm not sure if this condition is sufficient for you. The `old.Status.CurrentOperation` is being set at the beginning of reconciling the new generation. This means that:
1. User B could slip his changes before User A's changes are picked up by controller-manager
2. If the controller-manager is down (while apiserver is not), there could be a lot of spec changes from different users collected in the meantime.

The condition for locking should be not based on whether there is an operation in progress. It should be based on whether controller has finished processing the current generation (i.e. either succeeded, or failed and won't retry).

So I would suggest to change the condition to something more strict.
I suggest to copy the `isServiceInstanceProcessedAlready` method from controller
```go
if old.Generation != new.Generation && oldUID != newUID && !isServiceInstanceProcessedAlready(old) { ...
```
and copy (will also need to change `v1beta1` -> `sc` package)
https://github.com/kubernetes-incubator/service-catalog/blob/e15b73719911853d3755b71f3d8b26b21296d0a3/pkg/controller/controller_instance.go#L818-L826

P.S. I know that the condition was written this way before, but given that we decided to change it, it's worth fixing it as well.